docs/content/sips/007-deployment-auth.md

```diff
@@ -0,0 +1,158 @@
+
+It would be useful to have an unambiguous way of reporting this for programmatic consumers (e.g. distinguished error code, JSON output option), so they could display login UI or provide a specific prompt.  That may be a bit much to manage in the first drop though.
+
+## `spin login` detailed considerations
+
+`spin login` is a new command and a relatively complex user experience.  This section describes the behaviour in more detail.
+
+### Browser interaction
+
+As part of device auth flow, the user needs to go to the verification URL in their browser and paste in the code.
+
+> QUESTION: Should we attempt to open the browser automatically?
+> * Pro: it's more convenient
+> * Con: it may be unexpected (we have not asked for consent or warned it will happen)
+> * Con: it may work differently across browsers and operating systems, which may require explanation in the documentation
+
+> QUESTION: Should we copy the user code to the clipboard?
+> * Pro: it's convenient
+> * Con: the user may already have stuff on their clipboard
+> * Con: again, documentation needs to consider differences across OSes
+
+(For both of these questions, is there prior art to which we can refer?)
+
+### Verification polling
+
+The device auth flow involves the client polling the server using the device code as the auth token. Each poll can have three outcomes:
+
+* 200 and the response contains a token: user has authorised this device
+* 200 and the response doesn't contain a token: user has not yet authorised this device
+* Server error
+* Network error
+
+We propose to poll for two minutes. If the user has not entered the code by then, `spin login` fails and exits.
+
+Errors do _not_ cause polling to terminate. We do not want a transient network error or server hiccup to block login. That said, if we can identify server responses that are necessarily fatal, we can in future terminate early on those.
+
+> QUESTION: Are there any accessibility considerations around the timeout? For example, should we provide a way to override it (e.g. longer timeout, poll only when the user is ready) for users who might not be able to execute the process within two minutes? Is there prior art from other applications we can refer to here?
+
+### Token storage
+
+When the user logs into a server, we need to store:
+
+* The server URL
+* Any server connection options, such as the `--insecure` flag
+* The token
+
+A simple solution is to store these in a file under a non-roaming directory. This is not fantastically secure for tokens, but may be enough for the preview.
+
+> QUESTION: What priority should we assign to storing tokens securely?  (E.g. DPAPI on Windows, not sure of the equivalent across other OSes.)
+
+> QUESTION: Should we provide a way for the user to control where things are stored, opening up the possibility of multiple configurations? (This also provides an approach to the default/override URL question on deploy.)
```

Comment:
This seems like the simplest way to allow for multiple logins for "power users".